exercises/sinatra.md

```diff
@@ -0,0 +1,75 @@
+information for solving the following tasks:
+
+1. Create a new directory for this exercise (for example, use the name "sinatra-exercise"). Create a `Gemfile`. Add these lines to the `Gemfile`:
+    ```ruby
+source 'https://rubygems.org'
+gem 'sinatra'
+    ```
+    Then run `bundle install` in the directory to install sinatra.
+
+2. Next we will create a very basic web application using Sinatra as a starting point. Create a new file called app.rb and add the following to it:
+    ```ruby
+require 'sinatra'
+get '/' do
+  'Hello world!'
+end
+    ```
+
+3. Start the server by running `ruby app.rb` in your terminal. The program will start and continue running until you explicitly stop it.
+
+    It should look similar to this:
+    ![Sinatra starting in the command line](sinatra-start.png)
+
+    You can now use your browser to make a request to this web server running in your command line. To do so, enter the URL [http://localhost:4567](http://localhost:4567). Here's what you should see:
+
+    ![A browser window showing the text "Hello world!"](hello-world-browser.png)
+
+    **IMPORTANT**: From now on, whenever you make changes in app.rb, you will have to stop and restart your Sinatra application to see the changes. You can stop it with <kbd>Ctrl + C</kbd> and start by running `ruby app.rb` again.
+
+4. Let's add another route to our application:
+    ```ruby
+get '/curriculum' do
+    'This is our curriculum'
+end
+    ```
+
+    You can visit [http://localhost:4567/curriculum](http://localhost:4567/curriculum) to see the result. If you get an error message, then maybe you forgot to restart the app?
+
+5. Remember ERB? We can use it to generate web pages with Sinatra! Try changing the `curriculum` route so it uses an ERB template:
+    ```ruby
+get '/curriculum' do
+    erb :curriculum
+end
+    ```
+    Check the new results in your browser by refreshing the page. 
+    
+    ![Sinatra error message about a file not being found](sinatra-template-file-not-found.png)
+
+    Oh no! We get an error message! However, the message tells us exactly what is wrong: Sinatra is expecting the file `views/curriculum.erb` to exist in our directory.
+
+6. The `:curriculum` symbol refers to the name of the template file (`curriculum.erb`). Sinatra always expects ERB templates in the `view` directory, hence the path `views/curriculum.erb`. So let's create that file and fix the problem!
+
+    Use a `<h1>` element for the title of our curriculum page. Here's what it should look like:
+
+    ![Screen shot of a browser window of the Sinatra application showing the title "Ruby Monstas Curriculum"](curriculum-1.png)
+
+7. Next, we want to show a list of Ruby Monstas lessons. For this, we prepared a CSV file (remember those? ;)). You can download it [here](curriculum.csv). Have a look at the file in your text editor. It has just 3 columns: `number`, `lesson` and `description`.
```

Comment:
I'd suggest to decouple the csv task and introducing the instance variable. I'd maybe also show them first how to use the instance variables and then ask them to do it again in different use case.